### PR TITLE
linstor: Fix template volume missing on copy node

### PR DESCRIPTION
**Pull request by rp-**
### Description

A TODO was overseen and never implemented,
which could trigger the following bug:

If Linstor didn't create a resource (diskless or diskfull) on the cloudstack choosen node, it would not be able to copy the template data there, it even seems no error was
triggered and the new template file silently just became empty/corrupt.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [x] Major
- [ ] Minor
- [ ] Trivial

### How Has This Been Tested?

Local Linstor test cluster, fix is very isolated to Linstor primary storage

### CODE REVIEW

File: plugins/storage/volume/linstor/src/main/java/com/cloud/hypervisor/kvm/storage/LinstorStorageAdaptor.java [hunk 1]
```diff
@@ -197,6 +197,10 @@ public boolean deleteStoragePool(KVMStoragePool pool) {
         return deleteStoragePool(pool.getUuid());
     }
 
+    /**
+     * createPhysicalDisk will check if the resource wasn't yet created and do so, also it will make sure
+     * it is accessible from this node (MakeAvailable).
+     */
```
Comment (DaanHoogland) [inline plugins/storage/volume/linstor/src/main/java/com/cloud/hypervisor/kvm/storage/LinstorStorageAdaptor.java:203]: should this javadoc be on the interface?

File: plugins/storage/volume/linstor/src/main/java/com/cloud/hypervisor/kvm/storage/LinstorStorageAdaptor.java [hunk 2]
```diff
@@ -228,7 +232,11 @@ public KVMPhysicalDisk createPhysicalDisk(String name, KVMStoragePool pool, Qemu
                 null,
                 null);
 
-            // TODO make available on node
+            ResourceMakeAvailable rma = new ResourceMakeAvailable();
+            rma.diskful(true);
+            ApiCallRcList answers = api.resourceMakeAvailableOnNode(rscName, localNodeName, rma);
+            handleLinstorApiAnswers(answers,
+                String.format("Linstor: Unable to make resource %s available on node: %s", rscName, localNodeName));
```
Comment (DaanHoogland) [inline plugins/storage/volume/linstor/src/main/java/com/cloud/hypervisor/kvm/storage/LinstorStorageAdaptor.java:239]: separate method?

File: plugins/storage/volume/linstor/src/main/java/com/cloud/hypervisor/kvm/storage/LinstorStorageAdaptor.java (hunk 1 above)
Comment (rp-) [reply plugins/storage/volume/linstor/src/main/java/com/cloud/hypervisor/kvm/storage/LinstorStorageAdaptor.java:212]: I don't think so, as I guess other `createPhysicalDisk()` calls might just create the actual disk, but not check before if it is necessary.
Just wanted that to be documented somewhere, maybe I also got your question wrong and I should move the comment in the method?

Comment (DaanHoogland) [reply plugins/storage/volume/linstor/src/main/java/com/cloud/hypervisor/kvm/storage/LinstorStorageAdaptor.java:212]: by no means, if this is specialised behaviour for the Linstor plugin this javadoc is in the right place!

Comment (DaanHoogland) [reply plugins/storage/volume/linstor/src/main/java/com/cloud/hypervisor/kvm/storage/LinstorStorageAdaptor.java:212]: it just seemed something that all storage plugins are supposed to de ;)

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan LLtest

**Review by rohityadavcloud — APPROVED** [review]
LGTM - didn't testing, I trust @rp- has tested this as this change in linstor specific plugin

**Comment by rohityadavcloud** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Review by shwstppr — APPROVED** [review]
Code LGTM

**Comment by shwstppr** [comment]
@DaanHoogland are your concerns answered sufficiently?

**Comment by shwstppr** [comment]
Merging this based on test results

**Pull request merged**